feature: use specific image widths for posters and backdrops

Noor-Serry opened the pull request:
This commit updates the application to use different image widths for posters and backdrops.

- Posters will now use `BASE_IMAGE_URL_W500`.
- Backdrops will now use `BASE_IMAGE_URL_W300`.

This change is implemented across various DTOs and mappers to ensure consistency. The `MovieRemoteMapper` and `TvShowRemoteMapper` have been updated to accept an `isPoster` boolean to determine which image URL to use.

# Pull Request Template

## Description
Provide a clear and concise description of what this pull request does. Include the purpose and context for the changes.

---

## Changes Made
List the changes introduced in this pull request:

- change in movie and tv show mapper 
- change in the dto to put different images size
- change the base image url to :
baseImageUrlWidth500 = "https://image.tmdb.org/t/p/w500"
baseImageUrlWidth300= "https://image.tmdb.org/t/p/w300"
<img width="1080" height="2400" alt="Screenshot_20250725_195230" src="https://github.com/user-attachments/assets/59613c16-b237-4490-bd36-c957380c9036" />
![Uploading Screenshot_20250725_195253.png…]()

---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Commit:
feature: use specific image widths for posters and backdrops

This commit updates the application to use different image widths for posters and backdrops.

- Posters will now use `BASE_IMAGE_URL_W500`.
- Backdrops will now use `BASE_IMAGE_URL_W300`.

This change is implemented across various DTOs and mappers to ensure consistency. The `MovieRemoteMapper` and `TvShowRemoteMapper` have been updated to accept an `isPoster` boolean to determine which image URL to use.

The `build.gradle.kts` file has been updated to include these new build configuration fields.

saraashraf11 commented:
Assign yourself on PR and edit the template, please.

mohamedd-hassan reviewed (dismissed):
if you modified the build config make sure to modify the cicd to work with it

Commit:
refactor: use constant for image base URLs

This commit removes the image base URLs from `BuildConfig` and introduces a new `ImageBaseUrlsConstant` object to store them. This change centralizes the image base URL definitions, making them easier to manage and modify.

All DTOs and mappers that previously relied on `BuildConfig` for image URLs have been updated to use the new constant.
